engine/consensus/sealing/engine_test.go

```diff
@@ -184,15 +192,21 @@ func (s *SealingEngineSuite) TestMultipleProcessingItems() {
 		defer wg.Done()
 		for _, approval := range approvals {
 			err := s.engine.Process(channels.ReceiveApprovals, approverID, approval)
-			s.Require().NoError(err, "should process approval")
+			s.Require().NoError(err, "should process approval (trusted)")
 		}
 	}()
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		for _, approval := range responseApprovals {
-			err := s.engine.Process(channels.ReceiveApprovals, approverID, approval)
-			s.Require().NoError(err, "should process approval")
+		for _, resp := range responseApprovals {
+			//  convert wire → trusted *before* calling Process,
+			// so the queue only ever holds *flow.ResultApproval.
+			v, err := resp.ToInternal() // returns *flow.ApprovalResponse
```

Comment:
Can you verify this behavior with assertion? cleanup this doc and add assertion to make sure it's indeed the desired type there.